### PR TITLE
[pyproto] keep alive support

### PR DESCRIPTION
**Pull request by yunhaoling**
addressing issue: https://github.com/Azure/azure-sdk-for-python/issues/22298

### CODE REVIEW

File: uamqp/aio/_client_async.py [hunk 1]
```diff
@@ -176,6 +176,23 @@ async def _do_retryable_operation_async(self, operation, *args, **kwargs):
...
+        start_time = time.time()
+        try:
+            while self._connection and not self._shutdown:
+                current_time = time.time()
+                elapsed_time = (current_time - start_time)
+                if elapsed_time >= interval:
+                    _logger.info("Keeping %r connection alive. %r",
+                                 self.__class__.__name__,
+                                 self._connection._container_id)
+                    await asyncio.shield(self._connection._get_remote_timeout(current_time))
```
Comment (annatisch) [inline uamqp/aio/_client_async.py:190]: I don't think we need to shield this any more?
Now that we're no longer calling into C code - I think simply propagating an Cancelled error is probably fine?
Or was there another reason we were shielding it?

File: uamqp/client.py [hunk 2]
```diff
@@ -256,6 +276,10 @@ def open(self):
                 auth_timeout=self._auth_timeout
             )
             self._cbs_authenticator.open()
+        if self._keep_alive:
+                self._keep_alive_thread = threading.Thread(target=self._keep_alive_worker)
```
Comment (annatisch) [inline uamqp/client.py:280]: Double indented? :)

**Review by annatisch — APPROVED** [review]
Looks good - a few minor comments

File: uamqp/aio/_client_async.py (hunk 1 above)
Comment (yunhaoling) [reply uamqp/aio/_client_async.py:190]: nah.. I can't think of reasons to shield it, I copy-pasted from uamqp code.
removed it.

File: uamqp/client.py (hunk 2 above)
Comment (yunhaoling) [reply uamqp/client.py]: nice catch, removed

**Pull request merged**